packages/@aws-cdk/core/lib/cfn-mapping.ts

```diff
@@ -43,27 +62,50 @@ export class CfnMapping extends CfnRefElement {
    * @returns A reference to a value in the map based on the two keys.
    */
   public findInMap(key1: string, key2: string): string {
-    // opportunistically check that the key exists (if the key does not contain tokens)
-    if (!Token.isUnresolved(key1) && !(key1 in this.mapping)) {
-      throw new Error(`Mapping doesn't contain top-level key '${key1}'`);
+    let fullyResolved = false;
+    if (!Token.isUnresolved(key1)) {
+      if (!(key1 in this.mapping)) {
+        throw new Error(`Mapping doesn't contain top-level key '${key1}'`);
+      }
+      if (!Token.isUnresolved(key2)) {
+        if (!(key2 in this.mapping[key1])) {
+          throw new Error(`Mapping doesn't contain second-level key '${key2}'`);
+        }
+        fullyResolved = true;
+      }
     }
-
-    // opportunistically check that the second key exists (if the key does not contain tokens)
-    if (!Token.isUnresolved(key1) && !Token.isUnresolved(key2) && !(key2 in this.mapping[key1])) {
-      throw new Error(`Mapping doesn't contain second-level key '${key2}'`);
+    if (fullyResolved) {
+      if (this.lazy) {
+        return this.mapping[key1][key2];
+      }
+    } else {
+      this.lazyRender = true;
     }
-
     return Fn.findInMap(this.logicalId, key1, key2);
   }
 
   /**
    * @internal
    */
   public _toCloudFormation(): object {
-    return {
-      Mappings: {
-        [this.logicalId]: this.mapping,
-      },
-    };
+    if (this.lazy === undefined && !this.lazyRender) {
+      this.informLazyUse();
+    }
+    if (!this.lazy || (this.lazy && this.lazyRender)) {
+      return {
+        Mappings: {
+          [this.logicalId]: this.mapping,
+        },
+      };
+    } else {
+      return {};
+    }
+  }
+
+  private informLazyUse() {
+    if (!this.lazyInformed) {
+      Annotations.of(this).addInfo('Consider making this CfnMapping a lazy mapping by providing `lazy: true`: either no findInMap was called or every findInMap could be immediately resolved without using Fn::FindInMap');
```

Comment:
```suggestion
      Annotations.of(this).addWarning('Consider making this CfnMapping a lazy mapping by providing `lazy: true`: either no findInMap was called or every findInMap could be immediately resolved without using Fn::FindInMap');
```

Reply:
warning feels a bit intense here since they're not necessarily doing anything wrong and I don't think strict synthesis should fail in this case